backend/src/main/java/com/example/vehiclesservice/service/GarageServiceImpl.java

```diff
@@ -0,0 +1,29 @@
+package com.example.vehiclesservice.service;
+
+import com.example.vehiclesservice.api.Garage;
+import com.example.vehiclesservice.model.GarageEntity;
+import com.example.vehiclesservice.repository.GarageRepository;
+import lombok.AllArgsConstructor;
+import org.modelmapper.ModelMapper;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+@Service
+@AllArgsConstructor
+public class GarageServiceImpl implements GarageService{
+
+    private final GarageRepository garageRepository;
+    private final ModelMapper modelMapper = new ModelMapper();
+
+    @Override
+    public List<Garage> getAllGarages() {
+        List<GarageEntity> garages = garageRepository.findAll();
+        return mapList(garages, Garage.class);
+    }
+
+    private <S, T> List<T> mapList(List<S> source, Class<T> targetClass) {
```

Comment:
Maybe this method can be extracted as a static method somewhere else so it doesn't repeat across services.